core/lib/zksync_core/src/api_server/web3/state.rs

```diff
@@ -577,17 +622,18 @@ impl Filters {
     }
 
     /// Retrieves filter from the state.
-    pub fn get(&self, index: U256) -> Option<&TypedFilter> {
-        self.state.get(&index)
+    pub fn get_and_update_stats(&mut self, index: U256) -> Option<&TypedFilter> {
```

Comment:
Nit: It may make sense to clone `TypedFilter` right here (so, return `Option<TypedFilter>`); otherwise, a shared reference will borrow `self` mutably (this is determined by `&mut self`), which sort of defeats its purpose.